src/xercesc/validators/common/DFAContentModel.cpp

```diff
@@ -661,8 +662,15 @@ void DFAContentModel::buildDFA(ContentSpecNode* const curNode)
     //  in the fLeafCount member.
     //
     fLeafCount=countLeafNodes(curNode);
+    // Avoid integer overflow in below fLeafCount++ increment
+    if (fLeafCount > std::numeric_limits<unsigned int>::max() - 1)
```

Comment:
Should this be `XMLSize_t` (a.k.a. `size_t`) rather than `unsigned int` to match the type of `fLeafCount`?

Reply:
fLeafCount is actually declared as unsigned int at line 236 of DFAContentModel.hpp (which is already much bigger than what is reasonable in practice :-))

Reply:
Oops, I was looking at the wrong definition in src/xercesc/validators/common/ContentLeafNameTypeVector.hpp when I was scanning over the `git grep` output.  In that case, that all looks fine!